textdb/textdb-dataflow/src/main/java/com/google/re2j/PublicRE2.java

```diff
@@ -0,0 +1,36 @@
+package com.google.re2j;
+
+/**
+ * Wrapper class of re2j.RE2, this class includes regex parse flags <br>
+ * It also explains the purpose of wrapper classes in this package. <br>
+ * 
+ * <a href='github.com/google/re2j'>RE2J</a> is a Java port of RE2, a regular expression engine that runs in linear time. <br>
+ * See <a href='github.com/google/re2/wiki/Syntax'>RE2 syntax</a> for syntax accpeted by RE2. <br>
+ * <p>
+ * We use RE2J to parse a regular expression. <br>
+ * To get its abstract syntax tree, we need to access RE2J's non-public member variables. 
+ * RE2J's varaibles are declared at package-level, which means we need to be in the same package to access them. <br>
+ * See <a href='http://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html'>Java Access Control</a> for different Java access levels. <br>
+ * <br>
+ * We considered various solutions, including copying all RE2J's source code into our codebase, 
+ * having our modified fork of RE2J, or using Java reflection to change access level at runtime. 
+ * However, to keep our codebase clean, we don't want to copy or modify RE2J's code. <br>
+ * <br>
+ * Jianfeng Jia found a nice trick that allows us to access package-level variables.
```

Comment:
"Jianfeng Jia found a nice trick that allows us to access package-level variables" -> "The following solution allows us to access package-level variables."